### PR TITLE
Reset players remainingAir on respawn

### PR DESCRIPTION
**Pull request by ghost**
As mentioned in #1597, your air level did not reset if you drown.

Now you get a full breath during respawn.

### CODE REVIEW

File: Spigot-API-Patches/0167-Reset-players-remainingAir-on-respawn.patch [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+diff --git a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+index 7ac07ac07ac0..7ac07ac07ac0 100644
+--- a/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
++++ b/src/main/java/org/bukkit/event/player/PlayerRespawnEvent.java
+@@ -17,6 +17,7 @@ public class PlayerRespawnEvent extends PlayerEvent {
+         super(respawnPlayer);
+         this.respawnLocation = respawnLocation;
+         this.isBedSpawn = isBedSpawn;
++        respawnPlayer.setRemainingAir(300);
```
Comment (Spottedleaf) [inline Spigot-API-Patches/0167-Reset-players-remainingAir-on-respawn.patch]: Please add // Paper

**Comment by electronicboy** [comment]
Events are simple data objects, they shouldn't have any side effect from creating an instance of one, the fix here is for that to be reset along with the rest of the health levels for the entity

File: Spigot-API-Patches/0167-Reset-players-remainingAir-on-respawn.patch (hunk 1 above)
Comment (ghost) [reply Spigot-API-Patches/0167-Reset-players-remainingAir-on-respawn.patch]: I scrapped this. I will update this when I finish writing

**Comment by ghost** [comment]
All good now

**Review by aikar — APPROVED** [review]

**Review by aikar — APPROVED** [review]

**Pull request merged**